packages/react/src/components/DetailsList/DetailsRow.base.tsx

```diff
@@ -292,9 +299,11 @@ export class DetailsRowBase extends React.Component<IDetailsRowBaseProps, IDetai
         {showCheckbox && (
           <div role="gridcell" aria-colindex={1} data-selection-toggle={true} className={this._classNames.checkCell}>
             {onRenderCheck({
+              id: `${this._checkboxId}`,
               selected: isSelected,
               anySelected: isSelectionModal,
               'aria-label': checkButtonAriaLabel,
+              'aria-labelledby': `${this._checkboxId} ${this._rowHeaderId}`,
```

Comment:
Since IDs should be unique on the page, you should be able to use only one of these instead of a nested selector. (Or if the ID isn't unique on the page, that should be fixed.)

Reply:
`aria-labelledby` can take multiple ids that gives an element a more descriptive name. In this case i'm passing the checkbox id (whose name will be the aria-label, i.e: 'select row') and rowheader id (whose name will be the content of each rowHeader in a row, i.e: item 0") that way the name of the checkbox in the a11y tree will be “select row item 0” as opposed to just “select row”.

![image](https://user-images.githubusercontent.com/8649804/109726337-6b7af400-7b67-11eb-993a-d463625267f5.png)

I did need to generate `_rowHeaderId` in the same component (`DetailsRow`) I generated `_checkboxId` since I needed both for the `aria-labelledby` attribute and then passed `_rowHeaderId` to the `RowFields` component to make the connection and make this possible - should I not be doing it this way?

Reply:
Ah sorry, I hadn't realized that aria-labelledby could take two IDs. That's useful!